Cilsil/Services/CfgParserService.cs

```diff
@@ -88,50 +103,88 @@ private void ComputeMethodCfg(MethodDefinition method)
-                do
+                var exceptionHandlingBlock = exceptionHandler;
+                var exceptionHandlingBlockStartOffset = -1;
+                var exceptionHandlingBlockEndOffset = -1;
+                TypeReference catchType = null;
+                try
                 {
-                    var nextInstruction = programState.PopInstruction();
-                    // Checks if there is a node for the offset that we can reuse.
-                    (var nodeAtOffset, var excessiveVisits) =
-                        programState.GetOffsetNode(nextInstruction.Offset);
-                    if (nodeAtOffset != null)
-                    {
-                        programState.PreviousNode.Successors.Add(nodeAtOffset);
-                    }
-                    else if (excessiveVisits)
+                    switch (exceptionHandlingBlock.HandlerType)
                     {
-                        TimeoutMethodCount++;
-                        Log.WriteError("Translation timeout.");
-                        Log.RecordUnfinishedMethod(programState.Method.GetCompatibleFullName(),
-                                                   nextInstruction.RemainingInstructionCount());
-                        translationUnfinished = true;
-                        break;
+                        case ExceptionHandlerType.Catch:
+                            catchType = exceptionHandlingBlock.CatchType;
+                            exceptionHandlingBlockStartOffset = exceptionHandlingBlock.HandlerStart.Offset;
+                            exceptionHandlingBlockEndOffset = exceptionHandlingBlock.HandlerEnd.Offset;
+                            break;
+
+                        case ExceptionHandlerType.Finally:
+                            exceptionHandlingBlockStartOffset = exceptionHandlingBlock.HandlerStart.Offset;
+                            exceptionHandlingBlockEndOffset = exceptionHandlingBlock.HandlerEnd.Offset;
+                            break;
+
+                        case ExceptionHandlerType.Filter:
+                            // Example: catch (ArgumentException e) when (e.ParamName == "…")   
+                            // Adds associated try block node offsets to a hashset.
+                            catchType = programState.Method.Module.Import(typeof(System.Exception));
+                            exceptionHandlingBlockStartOffset = exceptionHandlingBlock.FilterStart.Offset;
+                            exceptionHandlingBlockEndOffset = exceptionHandlingBlock.HandlerEnd.Offset;
+                            break;
+
+                        case ExceptionHandlerType.Fault:
+                        // uncommon case: fault block
+                        // Example: fault {}
+                        default:
+                            break;
                     }
-                    else if (!InstructionParser.ParseCilInstruction(nextInstruction, programState))
+                }
+                catch (Exception)
```

Comment:
Avoid generic exceptions. Exceptions should be specifically typed and also be handled (at least logged)

Reply:
Actually -- how can an exception even possibly be thrown in this try block?

Reply:
It is possible. I had this to fix a crush on one of our big evaluated repos. 

Maybe it is a good idea to log it. Let me fix it.

Reply:
Done